ddtrace/ddtrace.go

```diff
@@ -33,6 +33,9 @@ type Tracer interface {
 	// Inject injects a span context into the given carrier.
 	Inject(context SpanContext, carrier interface{}) error
 
+	// InjectWithOptions injects a span context into the given carrier with options.
+	InjectWithOptions(context SpanContext, carrier interface{}, opts ...InjectionOption) error
```

Comment:
Wouldn't adding to the `SpanContext` also be updating a public interface and be a breaking change? That said, I would indeed love to make this change less impactful. I followed initial suggestions from the team to use the propagator injection and carrier interfaces but the constraints are limiting. It would be easier to do some things if `contrib/database/sql` could _pull_ data from the span context like the service name, env, version in addition to the span id, trace id and sampling priority. 

I'd be interested in chatting about how to potentially rethink how the new feature is implemented while offering the same functionality. I've been discovering the code base with this change but I still don't have the full context that you have.